compiler/noirc_evaluator/src/ssa/validation/mod.rs

```diff
@@ -41,110 +38,9 @@ struct Validator<'f> {
     range_checks: HashMap<ValueId, u32>,
 }
 
-#[derive(Debug)]
-enum PendingSignedOverflowOp {
-    AddOrSub { bit_size: u32, result: ValueId },
-    Mul { bit_size: u32, mul_result: ValueId, cast_result: Option<ValueId> },
-}
-
 impl<'f> Validator<'f> {
     fn new(function: &'f Function) -> Self {
-        Self { function, signed_binary_op: None, range_checks: HashMap::default() }
-    }
-
-    /// Validates that any checked signed add/sub/mul are followed by the appropriate instructions.
-    /// Signed overflow is many instructions but we validate up to the initial truncate.
-    ///
-    /// Expects the following SSA form for signed checked operations:
-    /// Add/Sub -> Truncate
-    /// Mul -> Cast -> Truncate
-    fn validate_signed_op_overflow_pattern(&mut self, instruction: InstructionId) {
```

Comment:
Could we salvage this validation method as a debug-only post-check for the expansion pass?

Reply:
Yeah, I think this is sensible.

Reply:
added in [aad9b03](https://github.com/noir-lang/noir/pull/9074/commits/aad9b036ffaaa806fe1d8d646f9541f82ffcef31)

Reply:
I've removed this again as it was failing on preexsiting unchecked ops (which wouldn't have an overflow check added)